data/types/common/inputtime.go

```diff
@@ -0,0 +1,52 @@
+package commontypes
+
+import (
+	"time"
+
+	"github.com/tidepool-org/platform/data"
+	"github.com/tidepool-org/platform/data/types"
+	"github.com/tidepool-org/platform/pointer"
+	"github.com/tidepool-org/platform/structure"
+)
+
+type InputTime struct {
+	InputTime *string `json:"inputTime,omitempty" bson:"inputTime,omitempty"`
+}
+
+func ParseInputTime(parser structure.ObjectParser) *InputTime {
+	if !parser.Exists() {
+		return nil
+	}
+	datum := NewInputTime()
+	parser.Parse(datum)
+	return datum
+}
+
+func NewInputTime() *InputTime {
+	return &InputTime{}
+}
+
+func (i *InputTime) Parse(parser structure.ObjectParser) {
+	i.InputTime = parser.String("inputTime")
+}
+
+func (i *InputTime) Validate(validator structure.Validator) {
+	timeValidator := validator.String("inputTime", i.InputTime)
+	timeValidator.AsTime(types.TimeFormat)
+}
+
+func (i *InputTime) Normalize(normalizer data.Normalizer) {
+	if i.InputTime != nil && *i.InputTime != "" {
```

Comment:
Input time should not be valid if it is an empty string.

Reply:
Yep
the time parser will make the job